fineract-provider/src/main/java/org/apache/fineract/infrastructure/security/filter/TenantAwareBasicAuthenticationFilter.java

```diff
@@ -138,9 +138,6 @@ protected void doFilterInternal(HttpServletRequest request, HttpServletResponse
                     }
 
                     if (!FIRST_REQUEST_PROCESSED) {
-                        final String baseUrl = request.getRequestURL().toString().replace(request.getPathInfo(), "/");
```

Comment:
Why do we remove this?

Reply:
I replaced it with that [line](https://github.com/apache/fineract/pull/3915/files#diff-7f872659757b85d5fcd50938abc4f805e15499f5e22026161dff8740fa4c4433R22).

I didn't fully understand why, but it's for backward compatibility and it should be covered with configuration.
@vidakovic told me to leave it this way.

Reply:
Adding only one place will not be enought. If someone is not using your custom module, they are loosing the baseUrl settings... :/ @vidakovic Am i missing something here?

Reply:
> Adding only one place will not be enought. If someone is not using your custom module, they are loosing the baseUrl settings...

I couldn't understand the full logic of any of them yet.

but, if we need to configure the URL inside the `application-v3.properties` file when using the custom module, and the other settings when not using it, then could the following implementation solve the problem?

- application-v3.properties line
``` properties
configuredBaseUrl=https://localhost:8443${FINERACT_SERVER_SERVLET_CONTEXT_PATH:/fineract-provider}
```

- add the following implementation to the condition block in `TenantAwareBasicAuthenticationFilter`:
``` java
String configuredBaseUrl = System.getProperty("configuredBaseUrl");

if (configuredBaseUrl != null && !configuredBaseUrl.isEmpty()) {
    baseUrl = configuredBaseUrl;
}
```

What do you think?

Reply:
Not really a fan of having hardcoded URL anywhere `https://localhost:8443${FINERACT_SERVER_SERVLET_CONTEXT_PATH:/fineract-provider}` but i am leaving to Aleks to decide on this one. I dont think leaving the `final String baseUrl = request.getRequestURL().toString().replace(request.getPathInfo(), "/");` would cause any harm, but i might be missing something,

Reply:
@adamsaghy ... we can improve that setting; I think there are spring built-in vars/properties for the host and the port... that way it won't be hardcoded. 

Note: there are 2 reasons to replace that piece of code. First it just plain doesn't work when we introduce Spring Web MVC. Second, application.properties should be the only source of truth (there might be more examples like this and should be treated the same, for another day).

Reply:
... and we need to move that property to the default application.properties file.

Reply:
How to use spring built-in vars?
I tried this but didn't work:
```properties
# FINERACT-914
server.forward-headers-strategy=framework
server.port=${FINERACT_SERVER_PORT:8443}
server.address=${FINERACT_SERVER_ADDRESS:localhost}
server.servlet.context-path=${FINERACT_SERVER_SERVLET_CONTEXT_PATH:/fineract-provider}
server.compression.enabled=${FINERACT_SERVER_COMPRESSION_ENABLED:true}

# backwards compatibility for 'org.apache.fineract.infrastructure.security.filter.TenantAwareBasicAuthenticationFilter
baseUrl=https://${server.address}:${server.port}${server.servlet.context-path}
```

and the checks failed as following:

Reply:
How to use spring built-in vars?
I tried this but didn't work:
```properties
# FINERACT-914
server.forward-headers-strategy=framework
server.port=${FINERACT_SERVER_PORT:8443}
server.address=${FINERACT_SERVER_ADDRESS:localhost}
server.servlet.context-path=${FINERACT_SERVER_SERVLET_CONTEXT_PATH:/fineract-provider}
server.compression.enabled=${FINERACT_SERVER_COMPRESSION_ENABLED:true}

# backwards compatibility for 'org.apache.fineract.infrastructure.security.filter.TenantAwareBasicAuthenticationFilter
baseUrl=https://${server.address}:${server.port}${server.servlet.context-path}
```

and the checks failed as following:
```
Run curl -f -k --retry 10 --retry-connrefused --connect-timeout 30 --retry-delay 30 https://localhost:8443/fineract-provider/actuator/health
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
curl: (35) error:0A000[12](https://github.com/Zeyad2003/fineract/actions/runs/10129914159/job/28010527803#step:8:13)6:SSL routines::unexpected eof while reading
Error: Process completed with exit code 35.
```

any thoughts?